packages/graphql-cli-validate-magento-pwa-queries/lib/index.js

```diff
@@ -0,0 +1,177 @@
+        context.spinner.succeed();
+
+        context.spinner.start(
+            "Validating project's queries against the schema..."
+        );
+        const report = validator.executeOnFiles(queryFiles);
+        context.spinner.succeed();
+
+        // Intentionally insert a blank line to the console for readability.
+        console.log();
+
+        // Report the results.
+        if (report.errorCount === 0) {
+            console.log(chalk.green('All queries are valid.'));
+
+            process.exit(exitCodes.SUCCESS);
+        } else {
+            console.warn(chalk.red('Found some invalid queries:'));
+
+            const formatter = validator.getFormatter();
+            console.log(formatter(report.results));
+
+            console.log(getErrorResolutionDetails());
+
+            process.exit(exitCodes.FAILURE);
+        }
+
+        process.exit(0);
+    } catch (e) {
+        const message = e.message ? e.message : e;
+        context.spinner.fail(`An error occurred:\n${message}`);
+
+        process.exit(exitCodes.FAILURE);
+    }
+}
+
+/**
+ * Supplies an object containing the arguments that this plugin supports.
+ */
+function getSupportedArguments(args) {
+    return args.options(plugin.SUPPORTED_ARGUMENTS);
+}
+
+/**
+ * Creates a linter configuration with rules based on the current
+ * clients and project.
+ */
+function getValidator({ clients, project }) {
+    const clientRules = clients.map(clientName => ({
+        env: clientName,
+        projectName: project
+    }));
+
+    const ruleDefinition = ['error', ...clientRules];
+
+    const linterConfiguration = {
+        parser: 'babel-eslint',
+        plugins: ['graphql'],
+        rules: {
+            'graphql/template-strings': ruleDefinition,
+            'graphql/no-deprecated-fields': ruleDefinition
+        },
+        useEslintrc: false
+    };
+
+    return new eslint.CLIEngine(linterConfiguration);
+}
+
+/**
+ * Returns a string that contains more specific information for the developer
+ * on their project version and how to resolve version incompatibility conflicts.
+ */
+function getErrorResolutionDetails() {
+    let versionSpecificDetails = '';
+
+    const packageVersion = process.env.npm_package_version;
+    if (packageVersion) {
+        versionSpecificDetails = `This Venia store is from PWA Studio version ${chalk.green(
+            packageVersion
+        )}.`;
+
+        // Attempt to figure out the Magento version(s) that match this PWA version.
+        const allPWAVersions = Object.keys(compatibilityDefinitions);
+        const pwaMatch = allPWAVersions.find(pwaVersion => {
+            return semver.satisfies(packageVersion, pwaVersion);
+        });
+        if (pwaMatch) {
+            const compatibleMagentoVersion = compatibilityDefinitions[pwaMatch];
+            versionSpecificDetails += `\nSome components in this PWA send GraphQL queries that are only compatible with version ${chalk.underline(
```

Comment:
```suggestion
            versionSpecificDetails += `\nSome components in PWA studio ${chalk.underline(chalk.red(packageVersion))} send GraphQL queries that are only compatible with version ${chalk.underline(
```
It can get confusing with the version numbers. Being a little explicit here reads better, IMO.